### ISSUE TITLE
cmake error

### ISSUE THREAD

**Issue by Luna-yu** [SeanNaren/warp-ctc#186]
(yu_tf_py) root@localhost:/home/yu/warp-ctc/build# cmake ..
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- broken
CMake Error at /usr/share/cmake-3.10/Modules/CMakeTestCCompiler.cmake:52 (message):
  The C compiler

    "/usr/bin/cc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /home/yu/warp-ctc/build/CMakeFiles/CMakeTmp
    
    Run Build Command:"/usr/bin/make" "cmTC_c7659/fast"
    /usr/bin/make -f CMakeFiles/cmTC_c7659.dir/build.make CMakeFiles/cmTC_c7659.dir/build
    make[1]: Entering directory '/home/yu/warp-ctc/build/CMakeFiles/CMakeTmp'
    Building C object CMakeFiles/cmTC_c7659.dir/testCCompiler.c.o
    /usr/bin/cc    -o CMakeFiles/cmTC_c7659.dir/testCCompiler.c.o   -c /home/yu/warp-ctc/build/CMakeFiles/CMakeTmp/testCCompiler.c
    cc: error trying to exec 'cc1': execvp: No such file or directory
    CMakeFiles/cmTC_c7659.dir/build.make:65: recipe for target 'CMakeFiles/cmTC_c7659.dir/testCCompiler.c.o' failed
    make[1]: *** [CMakeFiles/cmTC_c7659.dir/testCCompiler.c.o] Error 1
    make[1]: Leaving directory '/home/yu/warp-ctc/build/CMakeFiles/CMakeTmp'
    Makefile:126: recipe for target 'cmTC_c7659/fast' failed
    make: *** [cmTC_c7659/fast] Error 2
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:7 (project)

-- Configuring incomplete, errors occurred!
See also "/home/yu/warp-ctc/build/CMakeFiles/CMakeOutput.log".
See also "/home/yu/warp-ctc/build/CMakeFiles/CMakeError.log".

**Comment by zhangfengyo** [comment]
i have meet the same problem, and i have use this repo for a long time, but it occur today

**Comment by Luna-yu** [comment]
> i have meet the same problem, and i have use this repo for a long time, but it occur today

A few days ago, I had no problem. Today, I suddenly appeared. I found a lot of solutions, but I couldn't solve them

**Comment by zhangfengyo** [comment]
yes, why it occur?

**Comment by zhangfengyo** [comment]
i am curious. i use the warp-ctc for hundreds times, but it occur before i submit the project to my boss.

**Comment by zhangfengyo** [comment]
did you run it on colab? maybe some update is done by google

**Comment by Luna-yu** [comment]
I changed the GCC version because there was an error in the next make step. After that, cmake made an error

**Comment by aa-amory** [comment]
> did you run it on colab? maybe some update is done by google

Yes I have tried it on colab, it through below error 

/content/warp-ctc/build
[ 11%] Building NVCC (Device) object CMakeFiles/warpctc.dir/src/warpctc_generated_reduce.cu.o
nvcc fatal   : Unsupported gpu architecture 'compute_30'
CMake Error at warpctc_generated_reduce.cu.o.cmake:219 (message):
  Error generating
  /content/warp-ctc/build/CMakeFiles/warpctc.dir/src/./warpctc_generated_reduce.cu.o


CMakeFiles/warpctc.dir/build.make:70: recipe for target 'CMakeFiles/warpctc.dir/src/warpctc_generated_reduce.cu.o' failed
make[2]: *** [CMakeFiles/warpctc.dir/src/warpctc_generated_reduce.cu.o] Error 1
CMakeFiles/Makefile2:146: recipe for target 'CMakeFiles/warpctc.dir/all' failed
make[1]: *** [CMakeFiles/warpctc.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2

I have tried to uninstall the cuda and install lower version (9.2) but still issue there.
!apt-get --purge remove cuda nvidia* libnvidia-*
!dpkg -l | grep cuda- | awk '{print $2}' | xargs -n1 dpkg --purge
!apt-get remove cuda-*
!apt autoremove
!apt-get update
!wget https://developer.nvidia.com/compute/cuda/9.2/Prod/local_installers/cuda-repo-ubuntu1604-9-2-local_9.2.88-1_amd64 -O cuda-repo-ubuntu1604-9-2-local_9.2.88-1_amd64.deb
!dpkg -i cuda-repo-ubuntu1604-9-2-local_9.2.88-1_amd64.deb
!apt-key add /var/cuda-repo-9-2-local/7fa2af80.pub
!apt-get update
!apt-get install cuda-9.2

**Comment by zhangfengyo** [comment]
> > did you run it on colab? maybe some update is done by google
> 
> Yes I have tried it on colab, it through below error
> 
> /content/warp-ctc/build
> [ 11%] Building NVCC (Device) object CMakeFiles/warpctc.dir/src/warpctc_generated_reduce.cu.o
> nvcc fatal : Unsupported gpu architecture 'compute_30'
> CMake Error at warpctc_generated_reduce.cu.o.cmake:219 (message):
> Error generating
> /content/warp-ctc/build/CMakeFiles/warpctc.dir/src/./warpctc_generated_reduce.cu.o
> 
> CMakeFiles/warpctc.dir/build.make:70: recipe for target 'CMakeFiles/warpctc.dir/src/warpctc_generated_reduce.cu.o' failed
> make[2]: *** [CMakeFiles/warpctc.dir/src/warpctc_generated_reduce.cu.o] Error 1
> CMakeFiles/Makefile2:146: recipe for target 'CMakeFiles/warpctc.dir/all' failed
> make[1]: *** [CMakeFiles/warpctc.dir/all] Error 2
> Makefile:129: recipe for target 'all' failed
> make: *** [all] Error 2
> 
> I have tried to uninstall the cuda and install lower version (9.2) but still issue there.
> !apt-get --purge remove cuda nvidia* libnvidia-*
> !dpkg -l | grep cuda- | awk '{print $2}' | xargs -n1 dpkg --purge
> !apt-get remove cuda-*
> !apt autoremove
> !apt-get update
> !wget https://developer.nvidia.com/compute/cuda/9.2/Prod/local_installers/cuda-repo-ubuntu1604-9-2-local_9.2.88-1_amd64 -O cuda-repo-ubuntu1604-9-2-local_9.2.88-1_amd64.deb
> !dpkg -i cuda-repo-ubuntu1604-9-2-local_9.2.88-1_amd64.deb
> !apt-key add /var/cuda-repo-9-2-local/7fa2af80.pub
> !apt-get update
> !apt-get install cuda-9.2

i change warpCTC to pytorch ctc module, there is just a bit to be changed, and  the output was great too.

**Comment by aa-amory** [comment]
> i change warpCTC to pytorch ctc module, there is just a bit to be changed, and the output was great too.
could you please share the exact steps?

**Comment by aa-amory** [comment]
!pwd
%cd /content
!git clone --recursive https://github.com/StickCui/warp-ctc-pytorch.git
%cd /content/warp-ctc-pytorch
!sh make.sh install

still same issue 

/content/warp-ctc-pytorch
/content
/content/warp-ctc-pytorch
/content/warp-ctc-pytorch
Make core shared object of libwarp_ctc.so
-- cuda found TRUE
CMake Warning at CMakeLists.txt:48 (FIND_PACKAGE):
  By not providing "FindTorch.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Torch", but
  CMake did not find one.

  Could not find a package configuration file provided by "Torch" with any of
  the following names:

    TorchConfig.cmake
    torch-config.cmake

  Add the installation prefix of "Torch" to CMAKE_PREFIX_PATH or set
  "Torch_DIR" to a directory containing one of the above files.  If "Torch"
  provides a separate development package or SDK, be sure it has been
  installed.


-- Torch found Torch_DIR-NOTFOUND
-- Building shared library with GPU support
-- Configuring done
-- Generating done
-- Build files have been written to: /content/warp-ctc-pytorch/warpctc/core/build
[ 28%] Building NVCC (Device) object CMakeFiles/warpctc.dir/src/warpctc_generated_reduce.cu.o
[ 28%] Building NVCC (Device) object CMakeFiles/warpctc.dir/src/warpctc_generated_ctc_entrypoint.cu.o
nvcc fatal   : Unsupported gpu architecture 'compute_30'
CMake Error at warpctc_generated_reduce.cu.o.cmake:219 (message):
  Error generating
  /content/warp-ctc-pytorch/warpctc/core/build/CMakeFiles/warpctc.dir/src/./warpctc_generated_reduce.cu.o


CMakeFiles/warpctc.dir/build.make:70: recipe for target 'CMakeFiles/warpctc.dir/src/warpctc_generated_reduce.cu.o' failed
make[2]: *** [CMakeFiles/warpctc.dir/src/warpctc_generated_reduce.cu.o] Error 1
make[2]: *** Waiting for unfinished jobs....
nvcc fatal   : Unsupported gpu architecture 'compute_30'
CMake Error at warpctc_generated_ctc_entrypoint.cu.o.cmake:219 (message):
  Error generating
  /content/warp-ctc-pytorch/warpctc/core/build/CMakeFiles/warpctc.dir/src/./warpctc_generated_ctc_entrypoint.cu.o


CMakeFiles/warpctc.dir/build.make:63: recipe for target 'CMakeFiles/warpctc.dir/src/warpctc_generated_ctc_entrypoint.cu.o' failed
make[2]: *** [CMakeFiles/warpctc.dir/src/warpctc_generated_ctc_entrypoint.cu.o] Error 1
CMakeFiles/Makefile2:146: recipe for target 'CMakeFiles/warpctc.dir/all' failed
make[1]: *** [CMakeFiles/warpctc.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by zhangfengyo** [comment]
> > i change warpCTC to pytorch ctc module, there is just a bit to be changed, and the output was great too.
> > could you please share the exact steps?

maybe you can find a repo using pytorch'CTC, maybe what you should do is just adding a softmax function.
